crowdsourcehinter/static/js/src/crowdsourcehinter.js

```diff
@@ -0,0 +1,311 @@
+     */
+    function getHint(problemGradedEvent){
+        $(".crowdsourcehinter_block", element).show();
+        $.ajax({
+            type: "POST",
+            url: runtime.handlerUrl(element, 'get_hint'),
+            data: JSON.stringify({"submittedanswer": unescape(problemGradedEvent[0])}),
+            success: showHint
+        });
+    }
+
+    /**
+     * Start student hint rating/contribution. This will allow students to contribute new hints
+     * to the hinter as well as vote on the helpfulness of the first hint they received
+     * for the current problem. This function is called after the student answers
+     * the question correctly.
+     */
+    function startHintRating(){
+        $('.csh_correct', element).show();
+        $(".csh_hint_reveal", element).hide();
+        if($('.csh_hint_creation', element)){
+        //send empty data for ajax call because not having a data field causes error
+            $.ajax({
+                type: "POST",
+                url: runtime.handlerUrl(element, 'get_used_hint_answer_data'),
+                data: JSON.stringify({}),
+                success: setHintRatingUX
+            });
+        }
+    }
+
+    /**
+     * Check whether or not the question was correctly answered by the student.
+     * The current method of checking the correctness of the answer is very brittle
+     * since we simply look for a string within the problemGradedEventData.
+     * HACK
+     * @param problemGradedEventData is the data from problem_graded event.
+     */
+    function checkIsAnswerCorrect(problemGradedEventData){
+        if (problemGradedEventData[1].search(/class="correct/) === -1){
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * Check whether student answered the question correctly and call the appropriate
+     * function afterwards. Current method for determining correctness if very brittle.
+     * @param event_type, element are both unused but automatically passed
+     * @param data is generated by problem_graded event, contains status and data of the problem block
+     */
+    function onStudentSubmission(){ return function(event_type, data, element){
+        //search method of correctness of problem is brittle
+        if (checkIsAnswerCorrect(data)){
+            startHintRating();
+        } else { //if the submitted answer is incorrect
+            getHint(data);
+        }
+    }}
+
+    /**
+     * Set the target problem for which to listen for the problem_graded event. Set target to first
+     * problem block if no hinting element has been manually entered.
+     */
+    if(data.hinting_element == undefined || data.hinting_element == ''){
+        //contains workaround because the data-usage-id shows up with ";_" in place of "/" in lms
+        hintingElement = ($('.xblock[data-block-type="problem"]').first().attr('data-usage-id')).replace(/;_/g, '/');
```

Comment:
In this situation, the correct element of the problem xblock needs to be determined for logger.listen. If the hinting element isn't set manually, the data_usage_id field of the first problem type xblock is what should be used as the second argument for logger.listen. The data.hinting_element would need to be manually set if there are two problem xblocks on the same page with a hinter attached to each.

If you were asking about the necessity of "replace(/;_/g, '/')" at the end of that line, I think it's because some encoding/decoding(?) makes the data-usage-id show up differently when looking at it through lms and studio.